docs/versionhistory.rst

```diff
@@ -7,6 +7,9 @@ This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
 
 - Fixed acquring a lock twice in the same task on asyncio hanging instead of raising a
   ``RuntimeError`` (`#798 <https://github.com/agronholm/anyio/issues/798>`_)
+- Fixed ``self`` being different than the test's ``self`` for async instance method
+  fixtures in unittest-style tests
```

Comment:
It's not just unittest style tests, it also applies to class based pytest style tests